pkg/reconciler/resource/container.go

```diff
@@ -225,13 +225,17 @@ func resourcesFrom(object interface{}) (resources *corev1.ResourceRequirements)
 	return
 }
 
-func setResources(res *corev1.ResourceList, cpu, mem resource.Quantity) {
+func setResources(res *corev1.ResourceList, cpu, mem *resource.Quantity) {
```

Comment:
@antoineco I had to set these input types to pointers to distinguish when resource quantities are 0 and when they are not set, otherwise setting resource override for say CPU limit would automatically set the Memory limit to 0, while it should remain undefined.
And one last thing - deep derivative does not seem to react to resource overrides deletion on the desired state 🙁 Is there a way to force semantic comparison on override attributes match states "both ways"?

Reply:
Sure, changing the `resource` package to accept pointers is also OK if it doesn't support what we need.
I didn't know that a zero quantity would translate to literally `0`.

Reply:
Regarding the deep derivative comparison of resources, I'm a bit torn.
Of course we could define a custom comparison function for all the overridable fields (`corev1.ResourceList` and labels for the time being), but I'm wondering how often we expect people to 

1. Set requests/limits
1. Remove those requests/limits

I have a feeling people who do override requests/limits probably will **modify** those values (maybe), but I can't imagine a lot of cases where they will remove them.
If we think it's worth the effort, I'm all game for having an exception for `corev1.ResourceList`, but the case feels a bit edgy to me.

Reply:
I'm fine with leaving it as it is right now until someone reports it as an issue.